proto/pdpb.proto

```diff
@@ -668,3 +679,36 @@ message GetDCLocationInfoResponse {
     // which it's set when the number of the max suffix bits changes.
     Timestamp max_ts = 3;
 }
+
+message GetAllKeyspacesRequest {
+    RequestHeader header = 1;
+}
+
+message GetAllKeyspacesResponse {
+    ResponseHeader header = 1;
+    uint64 keyspace_version = 2;
+    // All registered keyspaces.
+    repeated keyspace.Keyspace keyspaces = 3;
+    // Prefixes that were registered to a keyspace and then unbound from it.
+    repeated keyspace.KeyspacePrefix unbound_prefixes = 4;
+}
+
+message PutKeyspaceRequest {
+    RequestHeader header = 1;
+    keyspace.Keyspace keyspace = 2;
+}
+
+message PutKeyspaceResponse {
+    ResponseHeader header = 1;
+    uint64 keyspace_version = 2;
```

Comment:
I don't think it's a good idea. To implement it correctly, you have to fetch the key space configuration again in case someone else has also made some updates.

Reply:
> you have to fetch the key space configuration again in case someone else has also made some updates

I can't see what conflicts with this.

Reply:
What's point of returning `keyspace_version`?